### PR TITLE
add support for m3u8 streams

### PR DESCRIPTION
**Pull request by KenanY**
Closes https://github.com/MemeLabs/Rustla2/issues/69
Closes https://github.com/MemeLabs/Rustla2/issues/77

### CODE REVIEW

File: api/src/ServicePoller.cpp [hunk 1]
```diff
@@ -32,6 +32,11 @@ void ServicePoller::Run() {
       status = CheckAngelThump(channel->GetChannel(), &state);
     } else if (channel->GetService() == kYouTubeService) {
       status = CheckYouTube(channel->GetChannel(), &state);
+    } else if (channel->GetService() == kM3u8Service) {
+      // Have m3u8 streams always appear as live for now.
+      stream->SetLive(true);
+      stream->Save();
+      return;
```
[the remark this answers is not in the document]
Comment (xDashh) [reply api/src/ServicePoller.cpp]: https://github.com/MemeLabs/Rustla2/issues/77 for reference.

**Comment by xDashh** [comment]
I'm not sure if this is done in this PR, but the custom m3u8 should probably contain the same client-side warning page as custom embeds (which shows the embed URL), since anyone could embed some random page in the m3u8 page and e.g. get chatters IPs via that.

File: api/src/ServicePoller.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (KenanY) [reply api/src/ServicePoller.cpp]: Yeah definitely; after sleeping on it I realized that I really should just implement `ServicePoller::CheckM3u8()` 👍

**Comment by KenanY** [comment]
@xDashh But the m3u8 URL goes straight into the clappr media player. Given that, do you still think there is a risk here?

**Comment by xDashh** [comment]
@KenanY I think so. All other embeds (as far as I'm aware) connect to some (somewhat) trusted 3rd party site, while in the custom embed and this the user can choose an arbitrary site.
We'd have to revisit the (security) considerations we made when initially implementing the "custom" service.

**Comment by slugalisk** [comment]
Yeah, he media player is lower risk than an iframe but - for example a malicious user could set up a web server at `http://my.memes/honeypot.m3u8` and log request IPs or set cookies that could be used to track users when they leave strims...

File: api/src/ServicePoller.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (KenanY) [reply api/src/ServicePoller.cpp]: Done.

**Comment by slugalisk** [comment]
@KenanY 👍

**Pull request merged**